azure-core/azure/core/pipeline/policies/universal.py

```diff
@@ -280,7 +280,12 @@ def deserialize_from_http_generics(cls, response):
         # Try to use content-type from headers if available
         content_type = None
         if 'content-type' in response.headers:
-            content_type = response.headers['content-type'].split(";")[0].strip().lower()
+            try:
+                content_type_header = response.headers['content-type'].split(";")[0]
+            except AttributeError:
+                content_type_header = response.headers['content-type'][0]
+            content_type = content_type_header.strip().lower()
```

Comment:
Might be worth extracting this block into a helper function

Reply:
I've moved this logic to the response object.